.gitignore

```diff
@@ -1,2 +1,40 @@
+.idea/
+
+# Compiled class file
+*.class
+
+# Log file
+*.log
+
+# BlueJ files
+*.ctxt
+
+# Mobile Tools for Java (J2ME)
+.mtj.tmp/
+
+# Package Files #
+*.jar
+*.war
+*.nar
+*.ear
+*.zip
+*.tar.gz
+*.rar
+
+# virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
+hs_err_pid*
+
 .gradle
-.idea
```

Comment:
You might want to tweak this by look at this stackoverflow post about what to ignore from intellij generated files. https://stackoverflow.com/questions/11968531/what-to-gitignore-from-the-idea-folder

Reply:
Also add these eclipse specific ones too.
```
.classpath
.project
.settings/
```